packages/client-search/src/types/Rule.ts

```diff
@@ -77,17 +77,29 @@ export type ConsequenceQuery = {
   }>;
 };
 
-export type ConsequencePromote = {
-  /**
-   * Unique identifier of the object to promote.
-   */
-  readonly objectID: string;
-
-  /**
-   * Promoted rank for the object (zero-based).
-   */
-  readonly position: number;
-};
+export type ConsequencePromote =
+  | {
+      /**
+       * Unique identifier of the object to promote.
+       */
+      readonly objectID: string;
+
+      /**
+       * Promoted rank for the object (zero-based).
+       */
+      readonly position: number;
+    }
+  | {
+      /**
+       * List of unique identifiers for the objects to promote.
+       */
+      readonly objectIDs: string[];
+
+      /**
+       * Promoted start rank for the objects (zero-based).
+       */
+      readonly position: number;
```

Comment:
in principle we could have this in a shared type, but I wanted a different description for position if matched with an array